File: VB/My/MyProject.vb
Imports Microsoft.VisualBasic
Imports Microsoft.VisualBasic.ApplicationServices
Imports Microsoft.VisualBasic.CompilerServices
Imports SuperImpactor
Imports System
Imports System.CodeDom.Compiler
Imports System.Collections
Imports System.ComponentModel
Imports System.ComponentModel.Design
Imports System.Diagnostics
Imports System.Reflection
Imports System.Runtime.CompilerServices
Imports System.Runtime.InteropServices
Imports System.Windows.Forms

Namespace SuperImpactor.My
	<GeneratedCode("MyTemplate", "11.0.0.0")>
	<HideModuleName>
	Friend Module MyProject
		Private ReadOnly m_ComputerObjectProvider As MyProject.ThreadSafeObjectProvider(Of MyComputer)

		Private ReadOnly m_AppObjectProvider As MyProject.ThreadSafeObjectProvider(Of MyApplication)

		Private ReadOnly m_UserObjectProvider As MyProject.ThreadSafeObjectProvider(Of Microsoft.VisualBasic.ApplicationServices.User)

		Private m_MyFormsObjectProvider As MyProject.ThreadSafeObjectProvider(Of MyProject.MyForms)

		Private ReadOnly m_MyWebServicesObjectProvider As MyProject.ThreadSafeObjectProvider(Of MyProject.MyWebServices)

		<HelpKeyword("My.Application")>
		Friend ReadOnly Property Application As MyApplication
			<DebuggerHidden>
			Get
				Return MyProject.m_AppObjectProvider.GetInstance
			End Get
		End Property

		<HelpKeyword("My.Computer")>
		Friend ReadOnly Property Computer As MyComputer
			<DebuggerHidden>
			Get
				Return MyProject.m_ComputerObjectProvider.GetInstance
			End Get
		End Property

		<HelpKeyword("My.Forms")>
		Friend ReadOnly Property Forms As MyProject.MyForms
			<DebuggerHidden>
			Get
				Return MyProject.m_MyFormsObjectProvider.GetInstance
			End Get
		End Property

		<HelpKeyword("My.User")>
		Friend ReadOnly Property User As Microsoft.VisualBasic.ApplicationServices.User
			<DebuggerHidden>
			Get
				Return MyProject.m_UserObjectProvider.GetInstance
			End Get
		End Property

		<HelpKeyword("My.WebServices")>
		Friend ReadOnly Property WebServices As MyProject.MyWebServices
			<DebuggerHidden>
			Get
				Return MyProject.m_MyWebServicesObjectProvider.GetInstance
			End Get
		End Property

		Sub New()
			MyProject.m_ComputerObjectProvider = New MyProject.ThreadSafeObjectProvider(Of MyComputer)()
			MyProject.m_AppObjectProvider = New MyProject.ThreadSafeObjectProvider(Of MyApplication)()
			MyProject.m_UserObjectProvider = New MyProject.ThreadSafeObjectProvider(Of Microsoft.VisualBasic.ApplicationServices.User)()
			MyProject.m_MyFormsObjectProvider = New MyProject.ThreadSafeObjectProvider(Of MyProject.MyForms)()
			MyProject.m_MyWebServicesObjectProvider = New MyProject.ThreadSafeObjectProvider(Of MyProject.MyWebServices)()
		End Sub

		<EditorBrowsable(EditorBrowsableState.Never)>
		<MyGroupCollection("System.Windows.Forms.Form", "Create__Instance__", "Dispose__Instance__", "My.MyProject.Forms")>
		Friend NotInheritable Class MyForms
			<ThreadStatic>
			Private Shared m_FormBeingCreated As Hashtable

			<EditorBrowsable(EditorBrowsableState.Never)>
			Public m_AppDetail As AppDetail

			<EditorBrowsable(EditorBrowsableState.Never)>
			Public m_AppIdDelete As AppIdDelete

			<EditorBrowsable(EditorBrowsableState.Never)>
			Public m_AppleAccounts As AppleAccounts

			<EditorBrowsable(EditorBrowsableState.Never)>
			Public m_AutoResign As AutoResign

			<EditorBrowsable(EditorBrowsableState.Never)>
			Public m_CertificateDelete As CertificateDelete

			<EditorBrowsable(EditorBrowsableState.Never)>
			Public m_CydiaRepoManager As CydiaRepoManager

			<EditorBrowsable(EditorBrowsableState.Never)>
			Public m_DownloadFiles As DownloadFiles

			<EditorBrowsable(EditorBrowsableState.Never)>
			Public m_DownloadProgress As DownloadProgress

			<EditorBrowsable(EditorBrowsableState.Never)>
			Public m_frmSuperImpactor As frmSuperImpactor

			<EditorBrowsable(EditorBrowsableState.Never)>
			Public m_Install As Install

			<EditorBrowsable(EditorBrowsableState.Never)>
			Public m_InstallResult As InstallResult

			<EditorBrowsable(EditorBrowsableState.Never)>
			Public m_Main As Main

			<EditorBrowsable(EditorBrowsableState.Never)>
			Public m_ReportBug As ReportBug

			<EditorBrowsable(EditorBrowsableState.Never)>
			Public m_UpdateCydia As UpdateCydia

			<EditorBrowsable(EditorBrowsableState.Never)>
			Public m_URLInstall As URLInstall

			Public Property AppDetail As AppDetail
				<DebuggerHidden>
				Get
					Me.m_AppDetail = MyProject.MyForms.Create__Instance__(Of AppDetail)(Me.m_AppDetail)
					Return Me.m_AppDetail
				End Get
				<DebuggerHidden>
				Set(ByVal value As AppDetail)
					If (value <> Me.m_AppDetail) Then
						If (value IsNot Nothing) Then
							Throw New ArgumentException("Property can only be set to Nothing")
						End If
						Me.Dispose__Instance__(Of AppDetail)(Me.m_AppDetail)
					End If
				End Set
			End Property

			Public Property AppIdDelete As AppIdDelete
				<DebuggerHidden>
				Get
					Me.m_AppIdDelete = MyProject.MyForms.Create__Instance__(Of AppIdDelete)(Me.m_AppIdDelete)
					Return Me.m_AppIdDelete
				End Get
				<DebuggerHidden>
				Set(ByVal value As AppIdDelete)
					If (value <> Me.m_AppIdDelete) Then
						If (value IsNot Nothing) Then
							Throw New ArgumentException("Property can only be set to Nothing")
						End If
						Me.Dispose__Instance__(Of AppIdDelete)(Me.m_AppIdDelete)
					End If
				End Set
			End Property

			Public Property AppleAccounts As AppleAccounts
				<DebuggerHidden>
				Get
					Me.m_AppleAccounts = MyProject.MyForms.Create__Instance__(Of AppleAccounts)(Me.m_AppleAccounts)
					Return Me.m_AppleAccounts
				End Get
				<DebuggerHidden>
				Set(ByVal value As AppleAccounts)
					If (value <> Me.m_AppleAccounts) Then
						If (value IsNot Nothing) Then
							Throw New ArgumentException("Property can only be set to Nothing")
						End If
						Me.Dispose__Instance__(Of AppleAccounts)(Me.m_AppleAccounts)
					End If
				End Set
			End Property

			Public Property AutoResign As AutoResign
				<DebuggerHidden>
				Get
					Me.m_AutoResign = MyProject.MyForms.Create__Instance__(Of AutoResign)(Me.m_AutoResign)
					Return Me.m_AutoResign
				End Get
				<DebuggerHidden>
				Set(ByVal value As AutoResign)
					If (value <> Me.m_AutoResign) Then
						If (value IsNot Nothing) Then
							Throw New ArgumentException("Property can only be set to Nothing")
						End If
						Me.Dispose__Instance__(Of AutoResign)(Me.m_AutoResign)
					End If
				End Set
			End Property

			Public Property CertificateDelete As CertificateDelete
				<DebuggerHidden>
				Get
					Me.m_CertificateDelete = MyProject.MyForms.Create__Instance__(Of CertificateDelete)(Me.m_CertificateDelete)
					Return Me.m_CertificateDelete
				End Get
				<DebuggerHidden>
				Set(ByVal value As CertificateDelete)
					If (value <> Me.m_CertificateDelete) Then
						If (value IsNot Nothing) Then
							Throw New ArgumentException("Property can only be set to Nothing")
						End If
						Me.Dispose__Instance__(Of CertificateDelete)(Me.m_CertificateDelete)
					End If
				End Set
			End Property

			Public Property CydiaRepoManager As CydiaRepoManager
				<DebuggerHidden>
				Get
					Me.m_CydiaRepoManager = MyProject.MyForms.Create__Instance__(Of CydiaRepoManager)(Me.m_CydiaRepoManager)
					Return Me.m_CydiaRepoManager
				End Get
				<DebuggerHidden>
				Set(ByVal value As CydiaRepoManager)
					If (value <> Me.m_CydiaRepoManager) Then
						If (value IsNot Nothing) Then
							Throw New ArgumentException("Property can only be set to Nothing")
						End If
						Me.Dispose__Instance__(Of CydiaRepoManager)(Me.m_CydiaRepoManager)
					End If
				End Set
			End Property

			Public Property DownloadFiles As DownloadFiles
				<DebuggerHidden>
				Get
					Me.m_DownloadFiles = MyProject.MyForms.Create__Instance__(Of DownloadFiles)(Me.m_DownloadFiles)
					Return Me.m_DownloadFiles
				End Get
				<DebuggerHidden>
				Set(ByVal value As DownloadFiles)
					If (value <> Me.m_DownloadFiles) Then
						If (value IsNot Nothing) Then
							Throw New ArgumentException("Property can only be set to Nothing")
						End If
						Me.Dispose__Instance__(Of DownloadFiles)(Me.m_DownloadFiles)
					End If
				End Set
			End Property

			Public Property DownloadProgress As DownloadProgress
				<DebuggerHidden>
				Get
					Me.m_DownloadProgress = MyProject.MyForms.Create__Instance__(Of DownloadProgress)(Me.m_DownloadProgress)
					Return Me.m_DownloadProgress
				End Get
				<DebuggerHidden>
				Set(ByVal value As DownloadProgress)
					If (value <> Me.m_DownloadProgress) Then
						If (value IsNot Nothing) Then
							Throw New ArgumentException("Property can only be set to Nothing")
						End If
						Me.Dispose__Instance__(Of DownloadProgress)(Me.m_DownloadProgress)
					End If
				End Set
			End Property

			Public Property frmSuperImpactor As frmSuperImpactor
				<DebuggerHidden>
				Get
					Me.m_frmSuperImpactor = MyProject.MyForms.Create__Instance__(Of frmSuperImpactor)(Me.m_frmSuperImpactor)
					Return Me.m_frmSuperImpactor
				End Get
				<DebuggerHidden>
				Set(ByVal value As frmSuperImpactor)
					If (value <> Me.m_frmSuperImpactor) Then
						If (value IsNot Nothing) Then
							Throw New ArgumentException("Property can only be set to Nothing")
						End If
						Me.Dispose__Instance__(Of frmSuperImpactor)(Me.m_frmSuperImpactor)
					End If
				End Set
			End Property

			Public Property Install As Install
				<DebuggerHidden>
				Get
					Me.m_Install = MyProject.MyForms.Create__Instance__(Of Install)(Me.m_Install)
					Return Me.m_Install
				End Get
				<DebuggerHidden>
				Set(ByVal value As Install)
					If (value <> Me.m_Install) Then
						If (value IsNot Nothing) Then
							Throw New ArgumentException("Property can only be set to Nothing")
						End If
						Me.Dispose__Instance__(Of Install)(Me.m_Install)
					End If
				End Set
			End Property

			Public Property InstallResult As InstallResult
				<DebuggerHidden>
				Get
					Me.m_InstallResult = MyProject.MyForms.Create__Instance__(Of InstallResult)(Me.m_InstallResult)
					Return Me.m_InstallResult
				End Get
				<DebuggerHidden>
				Set(ByVal value As InstallResult)
					If (value <> Me.m_InstallResult) Then
						If (value IsNot Nothing) Then
							Throw New ArgumentException("Property can only be set to Nothing")
						End If
						Me.Dispose__Instance__(Of InstallResult)(Me.m_InstallResult)
					End If
				End Set
			End Property

			Public Property Main As Main
				<DebuggerHidden>
				Get
					Me.m_Main = MyProject.MyForms.Create__Instance__(Of Main)(Me.m_Main)
					Return Me.m_Main
				End Get
				<DebuggerHidden>
				Set(ByVal value As Main)
					If (value <> Me.m_Main) Then
						If (value IsNot Nothing) Then
							Throw New ArgumentException("Property can only be set to Nothing")
						End If
						Me.Dispose__Instance__(Of Main)(Me.m_Main)
					End If
				End Set
			End Property

			Public Property ReportBug As ReportBug
				<DebuggerHidden>
				Get
					Me.m_ReportBug = MyProject.MyForms.Create__Instance__(Of ReportBug)(Me.m_ReportBug)
					Return Me.m_ReportBug
				End Get
				<DebuggerHidden>
				Set(ByVal value As ReportBug)
					If (value <> Me.m_ReportBug) Then
						If (value IsNot Nothing) Then
							Throw New ArgumentException("Property can only be set to Nothing")
						End If
						Me.Dispose__Instance__(Of ReportBug)(Me.m_ReportBug)
					End If
				End Set
			End Property

			Public Property UpdateCydia As UpdateCydia
				<DebuggerHidden>
				Get
					Me.m_UpdateCydia = MyProject.MyForms.Create__Instance__(Of UpdateCydia)(Me.m_UpdateCydia)
					Return Me.m_UpdateCydia
				End Get
				<DebuggerHidden>
				Set(ByVal value As UpdateCydia)
					If (value <> Me.m_UpdateCydia) Then
						If (value IsNot Nothing) Then
							Throw New ArgumentException("Property can only be set to Nothing")
						End If
						Me.Dispose__Instance__(Of UpdateCydia)(Me.m_UpdateCydia)
					End If
				End Set
			End Property

			Public Property URLInstall As URLInstall
				<DebuggerHidden>
				Get
					Me.m_URLInstall = MyProject.MyForms.Create__Instance__(Of URLInstall)(Me.m_URLInstall)
					Return Me.m_URLInstall
				End Get
				<DebuggerHidden>
				Set(ByVal value As URLInstall)
					If (value <> Me.m_URLInstall) Then
						If (value IsNot Nothing) Then
							Throw New ArgumentException("Property can only be set to Nothing")
						End If
						Me.Dispose__Instance__(Of URLInstall)(Me.m_URLInstall)
					End If
				End Set
			End Property

			<DebuggerHidden>
			<EditorBrowsable(EditorBrowsableState.Never)>
			Public Sub New()
				MyBase.New()
			End Sub

			<DebuggerHidden>
			Private Shared Function Create__Instance__(Of T As {Form, New})(ByVal Instance As T) As T
				Dim t1 As T
				If (If(Instance Is Nothing, False, Not Instance.IsDisposed)) Then
					t1 = Instance
				Else
					If (MyProject.MyForms.m_FormBeingCreated Is Nothing) Then
						MyProject.MyForms.m_FormBeingCreated = New Hashtable()
					ElseIf (MyProject.MyForms.m_FormBeingCreated.ContainsKey(GetType(T))) Then
						Throw New InvalidOperationException(Utils.GetResourceString("WinForms_RecursiveFormCreate", New String(-1) {}))
					End If
					MyProject.MyForms.m_FormBeingCreated.Add(GetType(T), Nothing)
					Try
						Try
							t1 = Activator.CreateInstance(Of T)()
						Catch targetInvocationException As System.Reflection.TargetInvocationException When targetInvocationException.InnerException IsNot Nothing
							Dim resourceString As String = Utils.GetResourceString("WinForms_SeeInnerException", New String() { targetInvocationException.InnerException.Message })
							Throw New InvalidOperationException(resourceString, targetInvocationException.InnerException)
						End Try
					Finally
						MyProject.MyForms.m_FormBeingCreated.Remove(GetType(T))
					End Try
				End If
				Return t1
			End Function

			<DebuggerHidden>
			Private Sub Dispose__Instance__(Of T As Form)(ByRef instance As T)
				instance.Dispose()
				instance = Nothing
			End Sub

			<EditorBrowsable(EditorBrowsableState.Never)>
			Public Overrides Function Equals(ByVal o As Object) As Boolean
				Return Me.Equals(RuntimeHelpers.GetObjectValue(o))
			End Function

			<EditorBrowsable(EditorBrowsableState.Never)>
			Public Overrides Function GetHashCode() As Integer
				Return Me.GetHashCode()
			End Function

			<EditorBrowsable(EditorBrowsableState.Never)>
			Friend Shadows Function [GetType]() As Type
				Return GetType(MyProject.MyForms)
			End Function

			<EditorBrowsable(EditorBrowsableState.Never)>
			Public Overrides Function ToString() As String
				Return Me.ToString()
			End Function
		End Class

		<EditorBrowsable(EditorBrowsableState.Never)>
		<MyGroupCollection("System.Web.Services.Protocols.SoapHttpClientProtocol", "Create__Instance__", "Dispose__Instance__", "")>
		Friend NotInheritable Class MyWebServices
			<DebuggerHidden>
			<EditorBrowsable(EditorBrowsableState.Never)>
			Public Sub New()
				MyBase.New()
			End Sub

			<DebuggerHidden>
			Private Shared Function Create__Instance__(Of T As New)(ByVal instance As T) As T
				Dim t1 As T
				t1 = If(instance IsNot Nothing, instance, Activator.CreateInstance(Of T)())
				Return t1
			End Function

			<DebuggerHidden>
			Private Sub Dispose__Instance__(Of T)(ByRef instance As T)
				instance = Nothing
			End Sub

			<DebuggerHidden>
			<EditorBrowsable(EditorBrowsableState.Never)>
			Public Overrides Function Equals(ByVal o As Object) As Boolean
				Return Me.Equals(RuntimeHelpers.GetObjectValue(o))
			End Function

			<DebuggerHidden>
			<EditorBrowsable(EditorBrowsableState.Never)>
			Public Overrides Function GetHashCode() As Integer
				Return Me.GetHashCode()
			End Function

			<DebuggerHidden>
			<EditorBrowsable(EditorBrowsableState.Never)>
			Friend Shadows Function [GetType]() As Type
				Return GetType(MyProject.MyWebServices)
			End Function

			<DebuggerHidden>
			<EditorBrowsable(EditorBrowsableState.Never)>
			Public Overrides Function ToString() As String
				Return Me.ToString()
			End Function
		End Class

		<ComVisible(False)>
		<EditorBrowsable(EditorBrowsableState.Never)>
		Friend NotInheritable Class ThreadSafeObjectProvider(Of T As New)
			<CompilerGenerated>
			<ThreadStatic>
			Private Shared m_ThreadStaticValue As T

			Friend ReadOnly Property GetInstance As T
				<DebuggerHidden>
				Get
					If (MyProject.ThreadSafeObjectProvider(Of T).m_ThreadStaticValue Is Nothing) Then
						MyProject.ThreadSafeObjectProvider(Of T).m_ThreadStaticValue = Activator.CreateInstance(Of T)()
					End If
					Return MyProject.ThreadSafeObjectProvider(Of T).m_ThreadStaticValue
				End Get
			End Property

			<DebuggerHidden>
			<EditorBrowsable(EditorBrowsableState.Never)>
			Public Sub New()
				MyBase.New()
			End Sub
		End Class
	End Module
End Namespace

File: super-impactor source/C#/My/MyProject.cs
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.ApplicationServices;
using Microsoft.VisualBasic.CompilerServices;
using SuperImpactor;
using System;
using System.CodeDom.Compiler;
using System.Collections;
using System.ComponentModel;
using System.ComponentModel.Design;
using System.Diagnostics;
using System.Reflection;
using System.Runtime.CompilerServices;
using System.Runtime.InteropServices;
using System.Windows.Forms;

namespace SuperImpactor.My
{
	[GeneratedCode("MyTemplate", "11.0.0.0")]
	[HideModuleName]
	internal static class MyProject
	{
		private readonly static MyProject.ThreadSafeObjectProvider<MyComputer> m_ComputerObjectProvider;

		private readonly static MyProject.ThreadSafeObjectProvider<MyApplication> m_AppObjectProvider;

		private readonly static MyProject.ThreadSafeObjectProvider<Microsoft.VisualBasic.ApplicationServices.User> m_UserObjectProvider;

		private static MyProject.ThreadSafeObjectProvider<MyProject.MyForms> m_MyFormsObjectProvider;

		private readonly static MyProject.ThreadSafeObjectProvider<MyProject.MyWebServices> m_MyWebServicesObjectProvider;

		[HelpKeyword("My.Application")]
		internal static MyApplication Application
		{
			[DebuggerHidden]
			get
			{
				return MyProject.m_AppObjectProvider.GetInstance;
			}
		}

		[HelpKeyword("My.Computer")]
		internal static MyComputer Computer
		{
			[DebuggerHidden]
			get
			{
				return MyProject.m_ComputerObjectProvider.GetInstance;
			}
		}

		[HelpKeyword("My.Forms")]
		internal static MyProject.MyForms Forms
		{
			[DebuggerHidden]
			get
			{
				return MyProject.m_MyFormsObjectProvider.GetInstance;
			}
		}

		[HelpKeyword("My.User")]
		internal static Microsoft.VisualBasic.ApplicationServices.User User
		{
			[DebuggerHidden]
			get
			{
				return MyProject.m_UserObjectProvider.GetInstance;
			}
		}

		[HelpKeyword("My.WebServices")]
		internal static MyProject.MyWebServices WebServices
		{
			[DebuggerHidden]
			get
			{
				return MyProject.m_MyWebServicesObjectProvider.GetInstance;
			}
		}

		static MyProject()
		{
			MyProject.m_ComputerObjectProvider = new MyProject.ThreadSafeObjectProvider<MyComputer>();
			MyProject.m_AppObjectProvider = new MyProject.ThreadSafeObjectProvider<MyApplication>();
			MyProject.m_UserObjectProvider = new MyProject.ThreadSafeObjectProvider<Microsoft.VisualBasic.ApplicationServices.User>();
			MyProject.m_MyFormsObjectProvider = new MyProject.ThreadSafeObjectProvider<MyProject.MyForms>();
			MyProject.m_MyWebServicesObjectProvider = new MyProject.ThreadSafeObjectProvider<MyProject.MyWebServices>();
		}

		[EditorBrowsable(EditorBrowsableState.Never)]
		[MyGroupCollection("System.Windows.Forms.Form", "Create__Instance__", "Dispose__Instance__", "My.MyProject.Forms")]
		internal sealed class MyForms
		{
			[ThreadStatic]
			private static Hashtable m_FormBeingCreated;

			[EditorBrowsable(EditorBrowsableState.Never)]
			public AppDetail m_AppDetail;

			[EditorBrowsable(EditorBrowsableState.Never)]
			public AppIdDelete m_AppIdDelete;

			[EditorBrowsable(EditorBrowsableState.Never)]
			public AppleAccounts m_AppleAccounts;

			[EditorBrowsable(EditorBrowsableState.Never)]
			public AutoResign m_AutoResign;

			[EditorBrowsable(EditorBrowsableState.Never)]
			public CertificateDelete m_CertificateDelete;

			[EditorBrowsable(EditorBrowsableState.Never)]
			public CydiaRepoManager m_CydiaRepoManager;

			[EditorBrowsable(EditorBrowsableState.Never)]
			public DownloadFiles m_DownloadFiles;

			[EditorBrowsable(EditorBrowsableState.Never)]
			public DownloadProgress m_DownloadProgress;

			[EditorBrowsable(EditorBrowsableState.Never)]
			public frmSuperImpactor m_frmSuperImpactor;

			[EditorBrowsable(EditorBrowsableState.Never)]
			public Install m_Install;

			[EditorBrowsable(EditorBrowsableState.Never)]
			public InstallResult m_InstallResult;

			[EditorBrowsable(EditorBrowsableState.Never)]
			public Main m_Main;

			[EditorBrowsable(EditorBrowsableState.Never)]
			public ReportBug m_ReportBug;

			[EditorBrowsable(EditorBrowsableState.Never)]
			public UpdateCydia m_UpdateCydia;

			[EditorBrowsable(EditorBrowsableState.Never)]
			public URLInstall m_URLInstall;

			public AppDetail AppDetail
			{
				[DebuggerHidden]
				get
				{
					this.m_AppDetail = MyProject.MyForms.Create__Instance__<AppDetail>(this.m_AppDetail);
					return this.m_AppDetail;
				}
				[DebuggerHidden]
				set
				{
					if (value != this.m_AppDetail)
					{
						if (value != null)
						{
							throw new ArgumentException("Property can only be set to Nothing");
						}
						this.Dispose__Instance__<AppDetail>(ref this.m_AppDetail);
					}
				}
			}

			public AppIdDelete AppIdDelete
			{
				[DebuggerHidden]
				get
				{
					this.m_AppIdDelete = MyProject.MyForms.Create__Instance__<AppIdDelete>(this.m_AppIdDelete);
					return this.m_AppIdDelete;
				}
				[DebuggerHidden]
				set
				{
					if (value != this.m_AppIdDelete)
					{
						if (value != null)
						{
							throw new ArgumentException("Property can only be set to Nothing");
						}
						this.Dispose__Instance__<AppIdDelete>(ref this.m_AppIdDelete);
					}
				}
			}

			public AppleAccounts AppleAccounts
			{
				[DebuggerHidden]
				get
				{
					this.m_AppleAccounts = MyProject.MyForms.Create__Instance__<AppleAccounts>(this.m_AppleAccounts);
					return this.m_AppleAccounts;
				}
				[DebuggerHidden]
				set
				{
					if (value != this.m_AppleAccounts)
					{
						if (value != null)
						{
							throw new ArgumentException("Property can only be set to Nothing");
						}
						this.Dispose__Instance__<AppleAccounts>(ref this.m_AppleAccounts);
					}
				}
			}

			public AutoResign AutoResign
			{
				[DebuggerHidden]
				get
				{
					this.m_AutoResign = MyProject.MyForms.Create__Instance__<AutoResign>(this.m_AutoResign);
					return this.m_AutoResign;
				}
				[DebuggerHidden]
				set
				{
					if (value != this.m_AutoResign)
					{
						if (value != null)
						{
							throw new ArgumentException("Property can only be set to Nothing");
						}
						this.Dispose__Instance__<AutoResign>(ref this.m_AutoResign);
					}
				}
			}

			public CertificateDelete CertificateDelete
			{
				[DebuggerHidden]
				get
				{
					this.m_CertificateDelete = MyProject.MyForms.Create__Instance__<CertificateDelete>(this.m_CertificateDelete);
					return this.m_CertificateDelete;
				}
				[DebuggerHidden]
				set
				{
					if (value != this.m_CertificateDelete)
					{
						if (value != null)
						{
							throw new ArgumentException("Property can only be set to Nothing");
						}
						this.Dispose__Instance__<CertificateDelete>(ref this.m_CertificateDelete);
					}
				}
			}

			public CydiaRepoManager CydiaRepoManager
			{
				[DebuggerHidden]
				get
				{
					this.m_CydiaRepoManager = MyProject.MyForms.Create__Instance__<CydiaRepoManager>(this.m_CydiaRepoManager);
					return this.m_CydiaRepoManager;
				}
				[DebuggerHidden]
				set
				{
					if (value != this.m_CydiaRepoManager)
					{
						if (value != null)
						{
							throw new ArgumentException("Property can only be set to Nothing");
						}
						this.Dispose__Instance__<CydiaRepoManager>(ref this.m_CydiaRepoManager);
					}
				}
			}

			public DownloadFiles DownloadFiles
			{
				[DebuggerHidden]
				get
				{
					this.m_DownloadFiles = MyProject.MyForms.Create__Instance__<DownloadFiles>(this.m_DownloadFiles);
					return this.m_DownloadFiles;
				}
				[DebuggerHidden]
				set
				{
					if (value != this.m_DownloadFiles)
					{
						if (value != null)
						{
							throw new ArgumentException("Property can only be set to Nothing");
						}
						this.Dispose__Instance__<DownloadFiles>(ref this.m_DownloadFiles);
					}
				}
			}

			public DownloadProgress DownloadProgress
			{
				[DebuggerHidden]
				get
				{
					this.m_DownloadProgress = MyProject.MyForms.Create__Instance__<DownloadProgress>(this.m_DownloadProgress);
					return this.m_DownloadProgress;
				}
				[DebuggerHidden]
				set
				{
					if (value != this.m_DownloadProgress)
					{
						if (value != null)
						{
							throw new ArgumentException("Property can only be set to Nothing");
						}
						this.Dispose__Instance__<DownloadProgress>(ref this.m_DownloadProgress);
					}
				}
			}

			public frmSuperImpactor frmSuperImpactor
			{
				[DebuggerHidden]
				get
				{
					this.m_frmSuperImpactor = MyProject.MyForms.Create__Instance__<frmSuperImpactor>(this.m_frmSuperImpactor);
					return this.m_frmSuperImpactor;
				}
				[DebuggerHidden]
				set
				{
					if (value != this.m_frmSuperImpactor)
					{
						if (value != null)
						{
							throw new ArgumentException("Property can only be set to Nothing");
						}
						this.Dispose__Instance__<frmSuperImpactor>(ref this.m_frmSuperImpactor);
					}
				}
			}

			public Install Install
			{
				[DebuggerHidden]
				get
				{
					this.m_Install = MyProject.MyForms.Create__Instance__<Install>(this.m_Install);
					return this.m_Install;
				}
				[DebuggerHidden]
				set
				{
					if (value != this.m_Install)
					{
						if (value != null)
						{
							throw new ArgumentException("Property can only be set to Nothing");
						}
						this.Dispose__Instance__<Install>(ref this.m_Install);
					}
				}
			}

			public InstallResult InstallResult
			{
				[DebuggerHidden]
				get
				{
					this.m_InstallResult = MyProject.MyForms.Create__Instance__<InstallResult>(this.m_InstallResult);
					return this.m_InstallResult;
				}
				[DebuggerHidden]
				set
				{
					if (value != this.m_InstallResult)
					{
						if (value != null)
						{
							throw new ArgumentException("Property can only be set to Nothing");
						}
						this.Dispose__Instance__<InstallResult>(ref this.m_InstallResult);
					}
				}
			}

			public Main Main
			{
				[DebuggerHidden]
				get
				{
					this.m_Main = MyProject.MyForms.Create__Instance__<Main>(this.m_Main);
					return this.m_Main;
				}
				[DebuggerHidden]
				set
				{
					if (value != this.m_Main)
					{
						if (value != null)
						{
							throw new ArgumentException("Property can only be set to Nothing");
						}
						this.Dispose__Instance__<Main>(ref this.m_Main);
					}
				}
			}

			public ReportBug ReportBug
			{
				[DebuggerHidden]
				get
				{
					this.m_ReportBug = MyProject.MyForms.Create__Instance__<ReportBug>(this.m_ReportBug);
					return this.m_ReportBug;
				}
				[DebuggerHidden]
				set
				{
					if (value != this.m_ReportBug)
					{
						if (value != null)
						{
							throw new ArgumentException("Property can only be set to Nothing");
						}
						this.Dispose__Instance__<ReportBug>(ref this.m_ReportBug);
					}
				}
			}

			public UpdateCydia UpdateCydia
			{
				[DebuggerHidden]
				get
				{
					this.m_UpdateCydia = MyProject.MyForms.Create__Instance__<UpdateCydia>(this.m_UpdateCydia);
					return this.m_UpdateCydia;
				}
				[DebuggerHidden]
				set
				{
					if (value != this.m_UpdateCydia)
					{
						if (value != null)
						{
							throw new ArgumentException("Property can only be set to Nothing");
						}
						this.Dispose__Instance__<UpdateCydia>(ref this.m_UpdateCydia);
					}
				}
			}

			public URLInstall URLInstall
			{
				[DebuggerHidden]
				get
				{
					this.m_URLInstall = MyProject.MyForms.Create__Instance__<URLInstall>(this.m_URLInstall);
					return this.m_URLInstall;
				}
				[DebuggerHidden]
				set
				{
					if (value != this.m_URLInstall)
					{
						if (value != null)
						{
							throw new ArgumentException("Property can only be set to Nothing");
						}
						this.Dispose__Instance__<URLInstall>(ref this.m_URLInstall);
					}
				}
			}

			[DebuggerHidden]
			[EditorBrowsable(EditorBrowsableState.Never)]
			public MyForms()
			{
			}

			[DebuggerHidden]
			private static T Create__Instance__<T>(T Instance)
			where T : Form, new()
			{
				T instance;
				if ((Instance == null ? false : !Instance.IsDisposed))
				{
					instance = Instance;
				}
				else
				{
					if (MyProject.MyForms.m_FormBeingCreated == null)
					{
						MyProject.MyForms.m_FormBeingCreated = new Hashtable();
					}
					else if (MyProject.MyForms.m_FormBeingCreated.ContainsKey(typeof(T)))
					{
						throw new InvalidOperationException(Utils.GetResourceString("WinForms_RecursiveFormCreate", new string[0]));
					}
					MyProject.MyForms.m_FormBeingCreated.Add(typeof(T), null);
					try
					{
						try
						{
							instance = Activator.CreateInstance<T>();
						}
						catch (TargetInvocationException targetInvocationException) when (targetInvocationException.InnerException != null)
						{
							string resourceString = Utils.GetResourceString("WinForms_SeeInnerException", new string[] { targetInvocationException.InnerException.Message });
							throw new InvalidOperationException(resourceString, targetInvocationException.InnerException);
						}
					}
					finally
					{
						MyProject.MyForms.m_FormBeingCreated.Remove(typeof(T));
					}
				}
				return instance;
			}

			[DebuggerHidden]
			private void Dispose__Instance__<T>(ref T instance)
			where T : Form
			{
				instance.Dispose();
				instance = default(T);
			}

			[EditorBrowsable(EditorBrowsableState.Never)]
			public override bool Equals(object o)
			{
				return this.Equals(RuntimeHelpers.GetObjectValue(o));
			}

			[EditorBrowsable(EditorBrowsableState.Never)]
			public override int GetHashCode()
			{
				return this.GetHashCode();
			}

			[EditorBrowsable(EditorBrowsableState.Never)]
			internal new Type GetType()
			{
				return typeof(MyProject.MyForms);
			}

			[EditorBrowsable(EditorBrowsableState.Never)]
			public override string ToString()
			{
				return this.ToString();
			}
		}

		[EditorBrowsable(EditorBrowsableState.Never)]
		[MyGroupCollection("System.Web.Services.Protocols.SoapHttpClientProtocol", "Create__Instance__", "Dispose__Instance__", "")]
		internal sealed class MyWebServices
		{
			[DebuggerHidden]
			[EditorBrowsable(EditorBrowsableState.Never)]
			public MyWebServices()
			{
			}

			[DebuggerHidden]
			private static T Create__Instance__<T>(T instance)
			where T : new()
			{
				T t;
				t = (instance != null ? instance : Activator.CreateInstance<T>());
				return t;
			}

			[DebuggerHidden]
			private void Dispose__Instance__<T>(ref T instance)
			{
				instance = default(T);
			}

			[DebuggerHidden]
			[EditorBrowsable(EditorBrowsableState.Never)]
			public override bool Equals(object o)
			{
				return this.Equals(RuntimeHelpers.GetObjectValue(o));
			}

			[DebuggerHidden]
			[EditorBrowsable(EditorBrowsableState.Never)]
			public override int GetHashCode()
			{
				return this.GetHashCode();
			}

			[DebuggerHidden]
			[EditorBrowsable(EditorBrowsableState.Never)]
			internal new Type GetType()
			{
				return typeof(MyProject.MyWebServices);
			}

			[DebuggerHidden]
			[EditorBrowsable(EditorBrowsableState.Never)]
			public override string ToString()
			{
				return this.ToString();
			}
		}

		[ComVisible(false)]
		[EditorBrowsable(EditorBrowsableState.Never)]
		internal sealed class ThreadSafeObjectProvider<T>
		where T : new()
		{
			[CompilerGenerated]
			[ThreadStatic]
			private static T m_ThreadStaticValue;

			internal T GetInstance
			{
				[DebuggerHidden]
				get
				{
					if (MyProject.ThreadSafeObjectProvider<T>.m_ThreadStaticValue == null)
					{
						MyProject.ThreadSafeObjectProvider<T>.m_ThreadStaticValue = Activator.CreateInstance<T>();
					}
					return MyProject.ThreadSafeObjectProvider<T>.m_ThreadStaticValue;
				}
			}

			[DebuggerHidden]
			[EditorBrowsable(EditorBrowsableState.Never)]
			public ThreadSafeObjectProvider()
			{
			}
		}
	}
}